### ISSUE TITLE
Can Ceresdb support data insert or modify time?

### ISSUE THREAD

**Issue by MachaelLee** [CeresDB/ceresdb#210]
### Describe This Problem

Can Ceresdb support data insert or modify time?

### Proposal

* In Oceanbase, I can record data insert time,  like gmt_create = now().

**Comment by Rachelint** [comment]
It is still not supported now , are you interested to make it?

**Comment by ShiKaiWi** [comment]
It's a very useful feature, and we will support it in the near future.

**Comment by ygf11** [comment]
Does it means we need support default value which is an `expr`?

The grammar may be like:
`CREATE TABLE test(a int DEFAULT b + 1, b int DEFAULT 0, c timestamp DEFAULT now())...;`

**Comment by ygf11** [comment]
> Yes, that gramma looks great! (only we don't support now() at present 🤣).

Let me have a try :D

**Comment by ShiKaiWi** [comment]
> `CREATE TABLE test(a int DEFAULT b + 1, b int DEFAULT 0, c timestamp DEFAULT now())`

In this statement, `a = b + 1` seems a little bit complex to implement, and I suppose the feature that defines default value as any expression) can be supported in another PR. What we can do here is to define the default value as some simple expression, e.g. literal value or some built-in function like `now`.

@ygf11 @waynexia What do you think of it?

**Comment by ygf11** [comment]
> In this statement, a = b + 1 seems a little bit complex to implement, and I suppose the feature that defines default value as any expression) can be supported in another PR. What we can do here is to define the default value as some simple expression, e.g. literal value or some built-in function like now.

It is ok to me. I will submit a pr today or tomorrow.

**Comment by ShiKaiWi** [comment]
> > In this statement, a = b + 1 seems a little bit complex to implement, and I suppose the feature that defines default value as any expression) can be supported in another PR. What we can do here is to define the default value as some simple expression, e.g. literal value or some built-in function like now.
> 
> I almost finish most of it, but submiting a simple implement is also ok to me. I will submit a pr today or tomorrow.

It will be surely great if you have made it.

**Comment by ygf11** [comment]
> I thought of one corner case when default value defined in expression

Yes, it is a problem. So circle reference is not allowed here, we can check and return error when create table.

> [MySQL](https://dev.mysql.com/doc/refman/8.0/en/data-type-defaults.html) only allow to reference a column defined before it, how will you avoid this?

We can visit and reorder the missing columns, the simple columns will run first, and other columns which all depends previous columns will run after.

for example:
```sql
CREATE TABLE t(c1 string tag not null, 
               ts timestamp not null, 
               c3 uint32 Default c4, 
               c4 uint32 Default c5, 
               c5 uint32 Default 0, timestamp key(ts),primary key(c1, ts)) \
               ENGINE=Analytic WITH (ttl='70d',update_mode='overwrite',arena_block_size='1KB')"

Insert into t(c1, ts) values(xx, xxx);
``` 

The original order of the missing column is `(c3, c4, c5)`.
After we reorder them, they will be `c5, c4 c3`. This running order is ok.

**Comment by ygf11** [comment]
> A tracking issue can be created to discuss those optimization.

I create a tracking issue #252.
